lte/gateway/c/oai/tasks/sgw/sgw_handlers.c

```diff
@@ -2242,10 +2243,21 @@ int sgw_handle_nw_initiated_actv_bearer_rsp(
     eps_bearer_ctxt_p->s_gw_teid_S1u_S12_S4_up =
       s11_actv_bearer_rsp->bearer_contexts.bearer_contexts[msg_bearer_index]
         .s1u_sgw_fteid.teid;
+    //S1U SGW IPv4 addr
+    eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.pdn_type = IPv4;
+    eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.address.ipv4_address
+        .s_addr = state->sgw_state.sgw_ip_address_S1u_S12_S4_up.s_addr;
+
     //S1U enb TEID
+    bstring bip = fteid_ip_address_to_bstring(
```

Comment:
why convert to bstring and back for a copy?

Reply:
Its just not covert and copy. fteid_t structure contains teid+ipv4/ipv6 address. fteid_ip_address_to_bstring() retrieves the ip address based on whether ipv6 or ipv6 bit is set.
bstring_to_ip_address() copies the ipv4/ipv6 addr to destination based on the length of the input string and also sets the pdn type

Reply:
Why not write another function to do just that instead of converting to string